### PR TITLE
It2

### PR DESCRIPTION
**Pull request by Tross0208**
Iteration 2 complete

### CODE REVIEW

File: lib/museum.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  def recommend_exhibits(patron)
+    recommendations = []
+    patron.interests.each do |interest|
+      @exhibits.each do |exhibit|
+        if interest == exhibit
+          recommendations << exhibit
+        end
+      end
+    end
+  end
```
Comment (cjsim89) [inline lib/museum.rb:21]: This method is not returning anything... It should end with `recommendations`

File: lib/museum.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+  def add_exhibit(exhibit)
+    @exhibits << exhibit
+  end
+
+  def recommend_exhibits(patron)
+    recommendations = []
+    patron.interests.each do |interest|
+      @exhibits.each do |exhibit|
+        if interest == exhibit
```
Comment (cjsim89) [inline lib/museum.rb:16]: In this context, `interest` is a String, and `exhibit` is an object of type Exhibit. So, this comparison will never return true. You may want to try doing `if interest == exhibit.name`, I think that's what you were going for?

File: spec/museum_spec.rb [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+    gems_and_minerals = Exhibit.new({name: "Gems and Minerals", cost: 0})
+    dead_sea_scrolls = Exhibit.new({name: "Dead Sea Scrolls", cost: 10})
+    imax = Exhibit.new({name: "IMAX",cost: 15})
+
+    dmns.add_exhibit(gems_and_minerals)
+    dmns.add_exhibit(dead_sea_scrolls)
+    dmns.add_exhibit(imax)
+
+    expect(dmns.exhibits).to be_instance_of(Array)
```
Comment (cjsim89) [inline spec/museum_spec.rb:36]: This test expects the same result as the test above that has **no** exhibits in the array. So, the test here is not thorough enough. It's good that it expects an `Array` but needs to be more specific - it should test that the array has exactly 3 indexes (in other words, that the Museum has 3 exhibits). Otherwise, the behavior of `add_exhibit` is not actually being tested at all.

File: spec/museum_spec.rb [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    gems_and_minerals = Exhibit.new({name: "Gems and Minerals", cost: 0})
+    dead_sea_scrolls = Exhibit.new({name: "Dead Sea Scrolls", cost: 10})
+    imax = Exhibit.new({name: "IMAX",cost: 15})
+
+    dmns.add_exhibit(gems_and_minerals)
+    dmns.add_exhibit(dead_sea_scrolls)
+    dmns.add_exhibit(imax)
+
+    expect(dmns.exhibits.length).to eq(3)
+  end
```
Comment (cjsim89) [inline spec/museum_spec.rb:67]: You have two of the same tests here; "has 3 exhibits"

File: spec/museum_spec.rb [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+    patron_1 = Patron.new("Bob", 20)
+    patron_1.add_interest("Dead Sea Scrolls")
+    patron_1.add_interest("Gems and Minerals")
+
+    patron_2 = Patron.new("Sally", 20)
+    patron_2.add_interest("IMAX")
+
+    expect(dmns.recommend_exhibits(patron_1).length).to eq(2)
+    expect(dmns.recommend_exhibits(patron_2).length).to eq(1)
+  end
```
Comment (cjsim89) [inline spec/museum_spec.rb:113]: You have two of the same test again, except they're worded slightly differently: "can recommend exhibits" and "recommends exhibits" test the same thing...? Maybe this is from a merge?

File: spec/museum_spec.rb [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+  it 'has name' do
+
+    dmns = Museum.new("Denver Museum of Nature and Science")
+    expect(dmns.name).to eq("Denver Museum of Nature and Science")
+  end
+
+  it 'can have exhibits' do
+
+    dmns = Museum.new("Denver Museum of Nature and Science")
+    expect(dmns.exhibits).to be_instance_of(Array)
```
Comment (cjsim89) [inline spec/museum_spec.rb:21]: Also test that this array is empty - either `dmns.exhibits.length == 0` or `emns.ehibits.empty? == true`. That way the test is thorough and given its context (has no exhibits assigned to the array yet, that's next) actually tests the expected behavior.

File: spec/museum_spec.rb (hunk 3 above)
Comment (cjsim89) [reply spec/museum_spec.rb:36]: To be clear, it is perfectly OK to have multiple `expect..` lines in your tests.
